### PR TITLE
deployment api checkin

### PR DESCRIPTION
**Pull request by jmspring**

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/functions/DeploymentToNodeMetadata.java [hunk 1]
```diff
@@ -35,48 +35,29 @@
...
+           ImmutableMap.<ComputeNode.Status, NodeMetadata.Status>builder().
+                   put(ComputeNode.Status.GOOD, NodeMetadata.Status.RUNNING).
+                   put(ComputeNode.Status.BAD, NodeMetadata.Status.ERROR).
+                   put(ComputeNode.Status.UNRECOGNIZED, NodeMetadata.Status.UNRECOGNIZED).
+                   build();
+
+   private static final Map<Deployment.ProvisioningState, NodeMetadata.Status> STATUS_TO_NODESTATUS =
+           ImmutableMap.<Deployment.ProvisioningState, NodeMetadata.Status>builder().
+                   put(Deployment.ProvisioningState.ACCEPTED, NodeMetadata.Status.PENDING).
+                   put(Deployment.ProvisioningState.READY, NodeMetadata.Status.PENDING).
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/functions/DeploymentToNodeMetadata.java:60]: How come `ACCEPTED` and `READY` are not mapped to `RUNNING` like the ASM implementation?

Comment (jmspring) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/functions/DeploymentToNodeMetadata.java:60]: The reason for this, at least in terms of a deployment, is that a Node is not running until the deployment has succeeded.  Accepted -> Azure has received the deployment request, Ready -> I believe is when the template deployment is ready to start.  I'm fine mirroring ASM, but I think with ARM, things are a little different.

File: azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/DeploymentApiLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                      "}";
+        return body;
+    }
+
+    @Test(groups = "live")
+    public void testCreate() {
+        Long now = System.currentTimeMillis();
+        resourceName = "jcres" + now;
+        deploymentName = "jcdep" + now;
+        String template = "{\"$schema\":\"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\"contentVersion\":\"1.0.0.0\",\"parameters\":{\"newStorageAccountName\":{\"type\":\"string\",\"metadata\":{\"description\":\"Name of the Storage Account\"}},\"storageAccountType\":{\"type\":\"string\",\"defaultValue\":\"Standard_LRS\",\"allowedValues\":[\"Standard_LRS\",\"Standard_GRS\",\"Standard_ZRS\"],\"metadata\":{\"description\":\"Storage Account type\"}},\"location\":{\"type\":\"string\",\"allowedValues\":[\"East US\",\"West US\",\"West Europe\",\"East Asia\",\"Southeast Asia\"],\"metadata\":{\"description\":\"Location of storage account\"}}},\"resources\":[{\"type\":\"Microsoft.Storage/storageAccounts\",\"name\":\"[parameters('newStorageAccountName')]\",\"apiVersion\":\"2015-05-01-preview\",\"location\":\"[parameters('location')]\",\"properties\":{\"accountType\":\"[parameters('storageAccountType')]\"}}]}";
```
Comment (ritazh) [inline azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/DeploymentApiLiveTest.java:62]: This template basically creates a storage account. Can we add more resources to this template to test an actual VM creation?

Comment (jmspring) [inline azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/DeploymentApiLiveTest.java:62]: I used a storage account to keep it simple and the tests to run reasonably quickly.  A full VM template can take minutes to fully deploy.  I'd say add a card to expand possible tests?

**Comment by ritazh** [comment]
LGTM

**Pull request merged**